overlord/snapstate/snapstate_update_test.go

```diff
@@ -12460,19 +12460,24 @@ type: snapd
 func (s *snapmgrTestSuite) setupSplitRefreshAppDependsOnModelBase(c *C, core18BasedApp bool) (names []string, infos []*snap.SideInfo, restore func()) {
 	restoreClassic := release.MockOnClassic(true)
 	modelRestore := snapstatetest.MockDeviceModel(ModelWithBase("core18"))
-	restore = func() {
-		restoreClassic()
-		modelRestore()
-	}
 
+	retryRestore := func() {}
```

Comment:
snapmgrTestSuite is likely already using testutil.BaseTest, which has AddCleanup(). It'd probably be too invasive for this PR, but the 'nop' restore like could be gracefully handled this way.

Reply:
That's a good idea, I've changed this now